wow/core/src/main/scala/wow/auth/AuthServer.scala

```diff
@@ -32,21 +31,22 @@ class AuthServer extends Actor with ActorLogging {
 
   initializeDatabase()
 
-  // TODO: should handlers be put in a pool so that they scale accordingly to the load ?
-  context.actorOf(LogonChallengeHandler.props, LogonChallengeHandler.PreferredName)
-  context.actorOf(LogonProofHandler.props, LogonProofHandler.PreferredName)
-  context.actorOf(ReconnectChallengeHandler.props, ReconnectChallengeHandler.PreferredName)
-  context.actorOf(ReconnectProofHandler.props, ReconnectProofHandler.PreferredName)
   context.actorOf(TCPServer.props(AuthSession, config.host, config.port), TCPServer.PreferredName)
 
-  // TODO: find a way to retrieve address and port
-  private val realms = Vector(ServerRealmlistEntry(1, 0, 0, "EnsiWoW", "127.0.0.1:8085", 0, 1, 1, 1))
-  private val serverRealmlistPacket: ServerRealmlist = ServerRealmlist(realms)
-  private val serverRealmlistPacketBits = PacketSerializer.serialize(serverRealmlistPacket)
-  private val eventRealmlist: EventRealmlist = EventRealmlist(serverRealmlistPacketBits)
+  private val realms = mutable.HashMap[Int, ServerRealmlistEntry]()
 
-  override def receive: PartialFunction[Any, Unit] = {
-    case GetRealmlist => sender() ! eventRealmlist
+  override def receive: Receive = {
+    case RegisterRealm(id, name, host, port) =>
+      log.debug(s"Realm added to list: $name at $host:$port")
+      realms(id) = ServerRealmlistEntry(1, 0, 0, name, s"$host:$port", 0, 1, 1, id)
+      AuthServer.realmlistPacketBits =
```

Comment:
The packet is the same for every player therefore re-serializing it every time is not necessary as we can just reuse the same bytes.
In the AuthServer, the packet is serialized every time a new realm is created. This should happen once or twice on startup at most.

For the future, this packet will be re-serialized every time the state of realm changes (i.e. if it crashes/is offline then the packet will be updated to reflect that) but we don't currently support that.